shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,97 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.CategoryDTO;
+import ro.msg.learning.shop.dto.ProductDTO;
+import ro.msg.learning.shop.entity.Product;
+import ro.msg.learning.shop.entity.ProductCategory;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.repo.ProductCategoryRepo;
+import ro.msg.learning.shop.repo.ProductRepo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+    private final ProductRepo productRepo;
+    private final ProductMapper productMapper;
+    private final ProductCategoryRepo productCategoryRepo;
+
+    public ProductDTO createProduct(ProductDTO productDto){
```

Comment:
have a look at this and see how you can integrate it in your project: https://www.baeldung.com/transaction-configuration-with-jpa-and-spring